app/src/features/apiClient/screens/apiClient/components/sidebar/APIClientSidebar.tsx

```diff
@@ -27,6 +28,16 @@ const APIClientSidebar: React.FC<Props> = () => {
   const [isNewRecordNameInputVisible, setIsNewRecordNameInputVisible] = useState(false);
   const [recordTypeToBeCreated, setRecordTypeToBeCreated] = useState<RQAPI.RecordType>();
 
+  const { addNewEnvironment, getAllEnvironments } = useEnvironmentManager();
+  const environments = getAllEnvironments();
+  const isGlobalEnvironmentExists = useMemo(() => environments.some((env) => env.isGlobal), [environments]);
+
+  useEffect(() => {
+    if (!isGlobalEnvironmentExists) {
```

Comment:
There is a race condition here. If i clear the storage and then refresh the page, it might lead to creation of multiple global environments
![CleanShot 2024-12-16 at 14 03 02@2x](https://github.com/user-attachments/assets/72156f25-d343-452d-b920-7be03fd94c97)